### PR TITLE
developed an image injection example

### PR DESCRIPTION
**Pull request by eyaly**

### CODE REVIEW

**Comment by wswebcreation** [comment]
Hi @eyaly 

I see a lot of code duplication, for the PO's and the test, can't this be simplified into 1 PO and 1 Test for both platforms? The App is written in such a way that it should support a cross-platform script

File: image-injection/src/test/java/pages/ios/SwagLabsPage.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.Set;
+
+public class SwagLabsPage {
+
+    By usernameID = By.id("test-Username");
+    By passwordID = By.id("test-Password");
+    By submitButton = By.id("test-LOGIN");
+    By ProductTitle = By.xpath("//XCUIElementTypeStaticText[@name=\"PRODUCTS\"]\n");
```
Comment (wswebcreation) [inline image-injection/src/test/java/pages/ios/SwagLabsPage.java:18]: Noooooo!!!

Please don't use XPATH for iOS, try `-ios predicate string` or `-ios class chain` to speed up the tests

Comment (eyaly) [reply image-injection/src/test/java/pages/ios/SwagLabsPage.java]: Thanks @wswebcreation . I changed the iOS XPath locators. Good feedback :-)

Comment (wswebcreation) [reply image-injection/src/test/java/pages/ios/SwagLabsPage.java]: You're welcome

**Comment by eyaly** [comment]
> Hi @eyaly
> 
> I see a lot of code duplication, for the PO's and the test, can't this be simplified into 1 PO and 1 Test for both platforms? The App is written in such a way that it should support a cross-platform script

Hi @wswebcreation 
For the long run, you are right, especially with the demo app that the business flow for iOS and Android is more or less the same. But still, the iOS and Android elements are not the same and we need 2 POs. Because of that, it is easy to demo with 2 tests - one for Android test and one for iOS test.

**Comment by wswebcreation** [comment]
> But still, the iOS and Android elements are not the same and we need 2 POs.

Hi @eyaly 

I don't completely agree with that. The only diff in the PO should be the locator, not the methods like for example sign in, verify a page is shown and so on. But again, I'm not the Java expert

**Comment by nadvolod** [comment]
> Hi @eyaly
> 
> I see a lot of code duplication, for the PO's and the test, can't this be simplified into 1 PO and 1 Test for both platforms? The App is written in such a way that it should support a cross-platform script



> > But still, the iOS and Android elements are not the same and we need 2 POs.
> 
> Hi @eyaly
> 
> I don't completely agree with that. The only diff in the PO should be the locator, not the methods like for example sign in, verify a page is shown and so on. But again, I'm not the Java expert

I agree with @wswebcreation here. Our Demo App supports a single PO model for ios and Android because Wim coded it that way :) Many customers don't have this luxury. However, you can solve this problem simply by removing the Page Objects. See my point above. But this is what I meant about introducing extra complexity for everyone. We wanted to show off image injection but now we are arguing page objects because what we are showing customers doesn't meet the SA standards. Since this is just an **example**, focus on the bare minimum to make this work :) If we want to include this as part of a framework then we would need to refactor the code to match what Wim suggested, plus include parallelization, TestWatcher and other framework best practices

**Comment by eyaly** [comment]
> > Hi @eyaly
> > I see a lot of code duplication, for the PO's and the test, can't this be simplified into 1 PO and 1 Test for both platforms? The App is written in such a way that it should support a cross-platform script
> 
> > > But still, the iOS and Android elements are not the same and we need 2 POs.
> > 
> > 
> > Hi @eyaly
> > I don't completely agree with that. The only diff in the PO should be the locator, not the methods like for example sign in, verify a page is shown and so on. But again, I'm not the Java expert
> 
> I agree with @wswebcreation here. Our Demo App supports a single PO model for ios and Android because Wim coded it that way :) Many customers don't have this luxury. However, you can solve this problem simply by removing the Page Objects. See my point above. But this is what I meant about introducing extra complexity for everyone. We wanted to show off image injection but now we are arguing page objects because what we are showing customers doesn't meet the SA standards. Since this is just an **example**, focus on the bare minimum to make this work :) If we want to include this as part of a framework then we would need to refactor the code to match what Wim suggested, plus include parallelization, TestWatcher and other framework best practices

@nadvolod - Is it better to create the structure as it is here:
https://github.com/saucelabs-training/demo-java/tree/master/appium-example/src/test/java/example

?

**Pull request merged**